opsdroid/connector/__init__.py

```diff
@@ -75,7 +130,25 @@ def configuration(self):
             bool: True for message successfully sent. False otherwise.
 
         """
-        raise NotImplementedError
+        warnings.warn(
```

Comment:
Should this be `_LOGGER.warning` instead? I'm mostly just asking here haha

Reply:
Hmmm, maybe, I don't know. Also is `DeprecationWarning` the best thing to use?

Reply:
To be fair the warnings.warn act similar to logging the warning, might be good to just keep to one way to log/warn users, perhaps Jacob has an opinion about this. I dont see a problem with `DeprecationWarning` 😄

Reply:
Well `DepreactionWarning` isn't printed by default, so people will only see it if it's logged by opsdroid somehow.

Reply:
@SolarDrew  added `warning.war` to message.py and the message is being logged, it seems that you can ignore these warnings so perhaps that is the reason why it was added?

Reply:
Yes, let's assume that was my reasoning.... Definitely not because of my limited understanding of the different behaviours of the two options...